yt_dlp/extractor/tv4.py

```diff
@@ -73,7 +73,7 @@ def _real_extract(self, url):
         video_id = self._match_id(url)
 
         info = self._download_json(
-            'https://playback-api.b17g.net/asset/%s' % video_id,
+            'https://playback2.a2d.tv/asset/%s' % video_id,
```

Comment:
use f-strings for both API URLs for consistency / readability
```suggestion
            f'https://playback2.a2d.tv/asset/{video_id}',
```

Reply:
Definitely make them consistent, but if it's done using f-strings yt-dl will just have to change it to any of the other myriad string syntaxes (I know that's our burden).

Maybe s/t like this:
```py
    def _call_api(self, endpoint, video_id, query=None, **kwargs):
        return self._download_json(
            '/'.join(('https://playback2.a2d.tv', endpoint, video_id)),
            video_id, endpoint.join(('Downloading video ', ' JSON')),
            query=update_url_query({
                'service': 'tv4',
                'device': 'browser',
                'protocol': 'hls,dash',
            }, query), **kwargs)
...
        info = self._call_api('asset', video_id, query={
                'drm': 'widevine',
            })['metadata']
...
        manifest_url = self._call_api('play', video_id, expected_status=401)
        # don't crash during error handling
        err = traverse_obj(manifest_url, 'errorCode')
...
        # then as in https://github.com/yt-dlp/yt-dlp/pull/5649#issuecomment-1337847672 below
```
Related to the geo-restriction handling, in the class:
```py
    # XFF is not effective
    _GEO_BYPASS = False
```
and perhaps specialise the `msg` for `raise_geo_restricted()` if, as I believe, a logged-in user can access the site from outside SE/EU:
```py
                self.raise_geo_restricted(
                    'This video is not available from your location due to geo restriction, or not being authenticated.',
                    countries=['SE'])
```

Reply:
> Definitely make them consistent, but if it's done using f-strings yt-dl will just have to change it to any of the other myriad string syntaxes (I know that's our burden).

Sorry, but if we discourage the use of newer Python features, then there was no point in deprecating older versions. In most cases (including here) fstrings simply read much better than the alternatives.

You could encourage OP to make PR against youtube-dl and get it merged there first - which would reverse the burdens; i.e. the burden of code review will then mainly fall on you, and the burden of porting (dealing with deprecated imports/functions) on me

Reply:
Fair point, but in this case it is at least possible to have just the one instance, and also avoid replicating the base API URL.

Reply:
oh, I wasn't commenting on the whole thing, just on the quoted part. Having a `_call_api` does look good